client/me/purchases/remove-plan-dialog/index.tsx

```diff
@@ -0,0 +1,156 @@
+import { Dialog, Gridicon } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { Fragment } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { SiteScreenshot } from '../site-screenshot';
+import getPlanFeatures from './get-plan-features';
+
+import './style.scss';
+
+interface RemovePlanDialogProps {
+	closeDialog: () => void;
+	removePlan: () => void;
+	isDialogVisible: boolean;
+	isRemoving: boolean;
+	site: SiteExcerptData;
+}
+
+export const RemovePlanDialog = ( {
+	closeDialog,
+	removePlan,
+	isDialogVisible,
+	site,
+}: RemovePlanDialogProps ) => {
+	const translate = useTranslate();
+
+	/**
+	 * Dialog buttons
+	 */
+	const buttons = [
+		{
+			action: 'cancel',
+			label: translate( 'Keep my plan' ),
+			onClick: closeDialog,
+		},
+		{
+			isPrimary: true,
+			action: 'removePlanAndAllSubscriptions',
+			label: translate( 'Cancel my plan', {
+				comment:
+					'This button removes the active plan and all active Marketplace subscriptions on the site',
+			} ),
+			onClick: removePlan,
+		},
+	];
+
+	/**
+	 * Determine if site date is within the first year
+	 */
+	const dateWithinLastYear = ( siteDate: string ) => {
+		const years =
+			new Date( new Date().getTime() - new Date( siteDate ).getTime() ).getFullYear() - 1970;
+
+		return years > 0 ? false : true;
+	};
+
+	/**
+	 * Istantiate site's plan variables.
+	 */
+	const productSlug = site.plan?.product_slug;
+	const planLabel = site.plan?.product_name_short;
+	const isComingSoon = site.is_coming_soon;
+	const isPrivate = site.is_private;
+	const launchedStatus = site.launch_status === 'launched' ? true : false;
+	const shouldUseSiteThumbnail =
+		isComingSoon === false && isPrivate === false && launchedStatus === true;
+	const withinFirstYear =
+		site.options.created_at !== undefined ? dateWithinLastYear( site.options.created_at ) : false;
+	const hasDomain = site.URL !== undefined && site.URL === site.options.unmapped_url ? false : true;
```

Comment:
This seems to be checking whether the site has a custom **primary** domain?  Doesn't seem like what you want here.  My guess is you want to know something more along the lines of whether they used their domain credit and still have the domain they used it on.

Reply:
This part is deprecated as the design changed and there are no more domains involved.